### PR TITLE
STAR-620 add sub navigation

### PR DESCRIPTION
**Pull request by woosie10**
This PR adds sub navigation as well as refactoring the existing navigation into smaller components.

The design is based on the following design, but kept more inline with our current style.
https://app.zeplin.io/project/5b18f693e66f3e4636a2a54a/screen/5cdea856407b951d80b99613

### CODE REVIEW

File: src/components/__tests__/__snapshots__/navigation_desktop.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+exports[`The NavigationDesktop component should render 1`] = `
+<div
+  className="navigation__desktop show-sm layout layout--gutter-none layout--align-middle"
+>
+  <div
+    className="layout__item u-fit-sm"
+  >
+    <nav
+      className="navigation__container"
+      role="navigation"
```
Comment (ImreBognarUltimaker) [inline src/components/__tests__/__snapshots__/navigation_desktop.test.tsx.snap]: It is not necessary to add navigation role to a nav element.
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Navigation_Role

Comment (woosie10) [reply src/components/__tests__/__snapshots__/navigation_desktop.test.tsx.snap]: updated

File: src/components/navigation_mobile.tsx [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    constructor(props) {
+        super(props);
+
+        this._onCloseMobileMenuHandler = this._onCloseMobileMenuHandler.bind(this);
+    }
+
+    private _onCloseMobileMenuHandler(): void {
+        const { onCloseMobileMenuHandler } = this.props;
+
+        if (onCloseMobileMenuHandler) {
```
Comment (ImreBognarUltimaker) [inline src/components/navigation_mobile.tsx]: The else of this if is not covered by the tests.

Comment (woosie10) [reply src/components/navigation_mobile.tsx]: added extra test

**Pull request merged**